data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java

```diff
@@ -136,6 +137,87 @@ public String getStringOrDefault(final String attribute, final String defaultVal
         throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
     }
 
+    /**
+     * Returns the value of the specified List<String>, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    @SuppressWarnings("unchecked")
+    public List<String> getStringListOrDefault(final String attribute, final List<String> defaultValue) {
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof List) {
+            ((List<?>) object).stream().filter(o -> !(o instanceof String)).forEach(o -> {
+                throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+            });
+            return (List<String>) object;
```

Comment:
This can be simplified:

```java
    ((List<?>) object).).forEach(o -> {
        if (!(o instanceof String)) {
                    throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
                }
```